.github/workflows/ci-angular.yaml

```diff
@@ -0,0 +1,80 @@
+name: Angular CI
+
+on:
+  push:
+    branches: [ "master" ]
+env:
+  IMAGE_NAME: angular
+  GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  
+jobs:
+  testing:
+     runs-on: ubuntu-latest
+     outputs:
+        get-job-id-test: ${{steps.job-id-test.outputs.get-job-id-test-out}}
+        short-sha1: ${{steps.env-short-sha.outputs.short-sha}}
+     strategy:
+         fail-fast: true
+     steps:
+     - name: Take Short SHA
+       id: env-short-sha
+       run: echo "::set-output name=short-sha::${GITHUB_SHA::7}"
+     - uses: actions/checkout@v1
+     - name: Use Node.js
+       uses: actions/setup-node@v1
+       with:
+        node-version:  '18.6'
+     - name: Test angular
+       run: |
+        npm install
+        npm run test -- --watch=false --browsers=ChromeHeadless
+       working-directory: angular
+     - name: Get Current ID Job 
+       uses: Tiryoh/gha-jobid-action@v0
+       id: get-job-id-test
+       with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          job_name: ${{github.job}}
+          per_page: 30
+     - name: Set job_id testing
+       id: job-id-test
+       run: echo "::set-output name=get-job-id-test-out::${{steps.get-job-id-test.outputs.job_id}}"
+  
+  build_push:
+    needs: [testing]
+    runs-on: ubuntu-latest
+    outputs:
+        get-job-id-build: ${{steps.job-id-build.outputs.get-job-id-build}}
+    strategy:
+      fail-fast: true
+    steps:
+    - name: Get Current ID Job 
+      uses: Tiryoh/gha-jobid-action@v0
+      id: get-job-id-build
+      with:
+        github_token: ${{ secrets.GITHUB_TOKEN }}
+        job_name: build_push
+        per_page: 30
+    - name: Set job_id build
+      id: job-id-build
+      run: echo "::set-output name=get-job-id-build::${{steps.get-job-id-build.outputs.job_id}}"
+    - uses: actions/checkout@v1
+    - name: Use Node.js 
+      uses: actions/setup-node@v1
+      with:
+        node-version: '18.6'
+    - name: Build angular
+      run: |
+        npm install
+        npm run build --if-present
+      working-directory: angular
+    - name: Login to GitHub Container Registry
+      run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u ${{ github.actor }} --password-stdin
+    - name: Build a container image from our angular app using Docker
+      run: docker build -t $IMAGE_NAME:${GITHUB_SHA::7} .
+    - name: Tag & publish to GitHub Container Registry
```

Comment:
did you remove the email notification?

Reply:
Yes, email notification will be sent by default to the person who  triggered start workflow (push commit). In order fir this to work, you need to go to the Settings -> Notifications -> In the actions section uncheck the box "Send notifications for failed workflows only"

Reply:
I know that :) but still need a custom implementation for the email delivery.

Reply:
Then return part of code with email notification? Last check you asked for  "Don't hardcode email details", but this action works with the help of authorization in gmail. Try to find other way for email notification?